pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/model/Agent.groovy

```diff
@@ -90,6 +90,23 @@ public class Agent extends MappedClosure<Object,Agent> implements Serializable {
         }
     }
 
+    /**
+     * Needed to handle the combination of describable ordinals *and* Descriptor lookup.
+     * @return The first symbol (in descriptor-ordinal-order searching) found in the map.
+     */
+    private String findSymbol() {
+        String sym = null
+        DeclarativeAgentDescriptor.all().each { d ->
```

Comment:
Well in this case I think a standard java for each loop found be more efficient since you then could break the loop when you have found the thing instead of doing the full combinatorics till the end.